resume-buddy/src/main/java/com/google/sps/servlets/BlobstoreServeServlet.java

```diff
@@ -0,0 +1,43 @@
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/blobstore-serve")
+public class BlobstoreServeServlet extends HttpServlet {
+  private BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Gets the logged in user's email, finds their blobKey and serves it
```

Comment:
Technically it serves the PDF blob, not the blob key. It passes the BlobKey to the BlobstoreService to tell it "hey, serve the blog associated with this BlobKey," and the BlobstoreService actually serves it into the response (the last line of this function).